src/pods/user-settings/user-settings.component.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import {
+  Typography,
+  Checkbox,
+  FormControlLabel,
+  Button,
+  TextField,
+  Container,
+} from '@material-ui/core';
+import { SettingsEntity } from 'core/settings';
+import produce from 'immer';
+
+interface Props {
+  onSave: (settings: SettingsEntity) => void;
+  onCancel: () => void;
+  userSettings: SettingsEntity;
+}
+
+export const UserSettingsComponent: React.FC<Props> = props => {
+  const { onSave, onCancel, userSettings } = props;
+  const [temporalSettings, setTemporalSettings] = React.useState<
+    SettingsEntity
+  >(userSettings);
+
+  React.useEffect(() => {
+    setTemporalSettings(userSettings);
+  }, [userSettings]);
+
+  const handleSecondChanceChange = () => (
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setTemporalSettings(
+      produce(temporalSettings, draft => {
+        draft.secondChance = !draft.secondChance;
+      })
```

Comment:
since this is an easy inmutable operation you can choose whether to use produce or spread operator, as you wish